packages/rich-text/src/component/event-listeners/index.js

```diff
@@ -26,13 +26,16 @@ const allEventListeners = [
 ];
 
 export function useEventListeners( props ) {
-	const propsRef = useRef( props );
+	const [ state, setState ] = useState( props );
```

Comment:
What's the warning here? This latest prop pattern should still be supported.

Reply:
Some unit tests are failing, so this code is wrong, and it seems to need to be reverted.

However, the latest `eslint-plugin-react-compiler` displays a warning for this code 🤔

![image](https://github.com/user-attachments/assets/d8263f08-c20a-46dd-a701-3bb049c24cfa)